device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -41,6 +41,7 @@
  * </p>
  * The client supports HTTPS 1.1 and AMQPS 1.0 transports.
  */
+@SuppressWarnings("deprecation") // A lot of references to deprecated APIs, but all of them are still in use
```

Comment:
I think I'm okay with this, but does this also suppress warnings when we use depreciated members from other packages?

Reply:
It would suppress them, yes. That's the one drawback to doing this

Reply:
It doesn't seem like we change the called APIs very often. I'm sure compilation or our tests will catch when a third party implementation breaks.